src/org/zaproxy/zap/extension/history/PopupMenuExportURLs.java

```diff
@@ -79,7 +79,19 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
         	    BufferedWriter fw = null;
                 try {
                     fw = new BufferedWriter(new FileWriter(file, isAppend));
-                    exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    int selectionCount=site.getSelectionCount();
+                    if (selectionCount == 0) {
+                        //No selection, export from root
+                        exportURLs((SiteNode) site.getModel().getRoot(), fw, html);
+                    } else if (selectionCount == 1) {

```

Comment:
It would also be cool if I could check what was selected in the Context section of the pane, then walk the site tree and see which items applied to the Context. (So that if a context is selected only the URLs applicable to it are exported...which would more completely address the original issue).

Is there an easy way to check for that?